### ISSUE TITLE
Implement `OR` logical operator for `QueryFilter`

### ISSUE THREAD

**Issue by RienNeVaPlus** [RienNeVaPlus/type-arango#23]
Below `v2.0.0` the `filter` feature of `entities.find(options)` uses the logical operator `AND`. Using `OR` is not supported.
```ts
const user: User[] = Users.find({filter:{type:'user',name:'bill'}})
// translates to: `FILTER i.type == "user" FILTER i.name == "bill"`
```
In order to support the `OR` operator, the default behavour in `v2.0.0` will change to using `OR` instead:
```ts
const user: User[] = Users.find({filter:{type:'user',name:'bill'}})
// translates to: `FILTER i.type == "user" OR i.name == "bill"`
```
Additionally an array of objects can be provided, each item represents a single filter statement - thus using the `AND` operator:
```ts
const user: User[] = Users.find({filter:[{type:'user'},{name:'bill'}]})
// translates to: `FILTER i.type == "user" FILTER i.name == "bill"`
```

Please keep this in mind when upgrading from `v1` to `v2`.

And to add some more confusion: `entities.find()` will be renamed to `entities.filter()` and `entities.findOne()` to `entities.find()` - as inteded.

**Comment by RienNeVaPlus** [comment]
Implemented in f4a69449aa1c269fd62960e3c26924cd64878e26

**Comment by RienNeVaPlus** [comment]
This was a stupid decision made with too little sleep. Sorry about the confusion. The operator should ofc always default to `AND`.
Instead, in `v2.0.2` a new property (`$`) can be used in order to provide a differen operator, [see docs](https://github.com/RienNeVaPlus/type-arango/blob/master/API.md#entitiesfilteroptions).
